doc/source/reference/routines.strings.rst

```diff
@@ -0,0 +1,58 @@
+String universal functions
+==========================
+
+.. currentmodule:: numpy.strings
+
+.. module:: numpy.strings
+
+The `numpy.strings` module provides a set of universal functions operating
+on arrays of type `numpy.str_`, `numpy.bytes_` or `StringDType`.
```

Comment:
Can you delete mentions of `StringDType` for now? I'll add them back in in the stringdtype PR.

Also unrelated but single backticks in rst are cross-references. You shouldn't use cross-references for symbols that aren't importable, for this one you wanted double backticks.